Fix install script and src/Makefile for EasyBuild.
Åke Sandgren, 2021-04-30
Update: Petr Král (INUITS)
Update: J. Saßmannshausen (Imperial College London/UK)
diff --git a/kraken2-2.14.orig/install_kraken2.sh b/kraken2-2.14/install_kraken2.sh
index bc65170..5d94edf 100755
--- a/kraken2-2.14.orig/install_kraken2.sh
+++ b/kraken2-2.14/install_kraken2.sh
@@ -23,7 +23,8 @@ fi
 
 # Perl cmd used to canonicalize dirname - "readlink -f" doesn't work
 # on OS X.
-export KRAKEN2_DIR=$(perl -MCwd=abs_path -le 'print abs_path(shift)' "$1")
+# export KRAKEN2_DIR=$(perl -MCwd=abs_path -le 'print abs_path(shift)' "$1")
+export KRAKEN2_DIR=$1
 
 mkdir -p "$KRAKEN2_DIR"
 make -C src install
diff --git a/kraken2-2.14.orig/src/Makefile b/kraken2-2.14/src/Makefile
index e6cb205..3479f37 100644
--- a/kraken2-2.14.orig/src/Makefile
+++ b/kraken2-2.14/src/Makefile
@@ -1,6 +1,6 @@
 CXX ?= g++
 KRAKEN2_SKIP_FOPENMP ?= -fopenmp
-CXXFLAGS = $(KRAKEN2_SKIP_FOPENMP) -Wall -std=c++11 -O3
+CXXFLAGS ?= $(KRAKEN2_SKIP_FOPENMP) -Wall -std=c++11 -O3
 CXXFLAGS += -DLINEAR_PROBING
 
 .PHONY: all clean install
